sdk/vppagent/client_connect.go

```diff
@@ -0,0 +1,106 @@
+package vppagent
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/ligato/vpp-agent/api/configurator"
+	"github.com/ligato/vpp-agent/api/models/vpp"
+	interfaces "github.com/ligato/vpp-agent/api/models/vpp/interfaces"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/local/connection"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/local/networkservice"
+	"github.com/networkservicemesh/networkservicemesh/sdk/common"
+	"github.com/networkservicemesh/networkservicemesh/sdk/endpoint"
+	"github.com/sirupsen/logrus"
+	"path"
+)
+
+// ClientConnect is a VPP Agent Client Connect composite
+type ClientConnect struct {
+	endpoint.BaseCompositeEndpoint
+	Workspace   string
+	Connections map[string]*ConnectionData
+}
+
+// Request implements the request handler
+func (cc *ClientConnect) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*connection.Connection, error) {
+
+	if cc.GetNext() == nil {
+		logrus.Fatal("The VPP Agent Client Connect composite requires that there is Next set")
+	}
+
+	incomingConnection, err := cc.GetNext().Request(ctx, request)
+	if err != nil {
+		logrus.Errorf("Next request failed: %v", err)
+		return nil, err
+	}
+
+	opaque := cc.GetNext().GetOpaque(incomingConnection)
+	if opaque == nil {
+		err := fmt.Errorf("received empty data from Next")
+		logrus.Errorf("Unable to find the outgoing connection: %v", err)
+		return nil, err
+	}
+	outgoingConnection := opaque.(*connection.Connection)
+	incomingConnection.Context = outgoingConnection.GetContext()
```

Comment:
Not sure when this should be done. Only ClientConnect has access to the outgoing connection object, but it looks more like cross connect stuff, doesn't it?